### ISSUE TITLE
NoMethodError: undefined method 'at_xpath' for nil:NilClass

### ISSUE THREAD

**Issue by guguso** [woahdae/simple_xlsx_reader#23]
Hi,
I have some troubles while parsing a file since yesterday. The file opens well on Microsoft Excel.
Should a new version of Microsoft Excel cause that bug?
```
NoMethodError: undefined method `at_xpath' for nil:NilClass
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:192:in `last_cell_label'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:204:in `sheet_dimensions'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:127:in `parse_sheet'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:109:in `block in load_sheets'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `each'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `each_with_index'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `each'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `map'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `load_sheets'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:40:in `sheets'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:29:in `tap'
/Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:29:in `open'
```

This bug appears when I try to parse the file BulletinSearch.xlsx from http://www.microsoft.com/en-us/download/details.aspx?id=36982

Thanks for help

**Comment by woahdae** [comment]
Not sure, the xlsx spec has a lot of features. Could you send along the
offending document? At first glance looks like it can't find a cell where
it thinks it is.

On Wednesday, September 9, 2015, Augustin G <notifications@github.com>
wrote:

> Hi,
> I have some troubles while parsing a file since yesterday. The file opens
> well on Microsoft Excel.
> Should a new version of Microsoft Excel cause that bug?
>
> NoMethodError: undefined method `at_xpath' for nil:NilClass
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:192:in `last_cell_label'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:204:in `sheet_dimensions'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:127:in `parse_sheet'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:109:in `block in load_sheets'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `each'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `each_with_index'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `each'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `map'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:108:in `load_sheets'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:40:in `sheets'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:29:in `tap'
> /Users/me/.rvm/gems/ruby-2.2.0/gems/simple_xlsx_reader-1.0.2/lib/simple_xlsx_reader.rb:29:in `open'
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/woahdae/simple_xlsx_reader/issues/23>.
>


-- 
Sent from my phone

**Comment by guguso** [comment]
This bug appears when I try to parse the file BulletinSearch.xlsx from http://www.microsoft.com/en-us/download/details.aspx?id=36982

Thanks for help

**Comment by steveny** [comment]
The issue appears to be in certain documents that if there is only one sheet the sheet is in xl/worksheets/sheet.xml instead of xl/worksheets/sheet1.xml.

**Comment by woahdae** [comment]
Thanks! I'll see if I can repro it in a test.

**Comment by steveny** [comment]
I know it is bad github form, but I did fork this and committed a change to master:

https://github.com/missiondata/simple_xlsx_reader/commit/b9e7c051a4cd1a297093a59b9d860aa631e45c52

Steven

**Comment by woahdae** [comment]
Cool! Wanna add a test case?

**Comment by woahdae** [comment]
Resolved by #24
